src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java

```diff
@@ -0,0 +1,109 @@
+
+  public TutorialAction() {
+    this(PluginSettings.getInstance(),
+         new DefaultNotifier());
+  }
+
+  public TutorialAction(@NotNull MainViewModelProvider mainViewModelProvider,
+                        @NotNull Notifier notifier) {
+    this.mainViewModelProvider = mainViewModelProvider;
+    this.notifier = notifier;
+  }
+
+  @Override
+  public void actionPerformed(@NotNull AnActionEvent e) {
+    if (e.getProject() == null) {
+      return;
+    }
+    confirmStart(e.getProject());
+  }
+
+  private void confirmStart(@NotNull Project project) {
+    MainViewModel mainViewModel = mainViewModelProvider.getMainViewModel(project);
+    CourseViewModel courseViewModel = mainViewModel.courseViewModel.get();
+    ExercisesTreeViewModel exercisesViewModel = mainViewModel.exercisesViewModel.get();
+    Authentication authentication = mainViewModel.authentication.get();
+
+    if (courseViewModel == null || exercisesViewModel == null || authentication == null) {
+      return;
+    }
+
+    BaseTreeViewModel.Selection selection = exercisesViewModel.findSelected();
+    ExerciseViewModel selectedExercise = (ExerciseViewModel) selection.getLevel(2);
+    ExerciseGroupViewModel selectedExerciseGroup = (ExerciseGroupViewModel) selection.getLevel(1);
+    if (selectedExercise == null || selectedExerciseGroup == null
+            || !ExerciseViewModel.Status.TUTORIAL.equals(selectedExercise.getStatus())) {
+      notifier.notifyAndHide(new ExerciseNotSelectedNotification(), project);
+      return;
+    }
+    Tutorial tutorial = ((TutorialExercise) selectedExercise.getModel()).getTutorial();
+    TutorialViewModel tutorialViewModel = new TutorialViewModel(tutorial,
+            TaskView::createAndShow, project);
+    mainViewModelProvider.getMainViewModel(project).tutorialViewModel.set(tutorialViewModel);
+    CompleteTutorial completeTutorial = new CompleteTutorial(mainViewModel);
+    tutorial.tutorialUpdated.addListener(completeTutorial, CompleteTutorial::completeTutorial);
+    StartTutorialDialog.createAndShow(tutorialViewModel);
+  }
+
+  @Override
+  public void update(@NotNull AnActionEvent e) {
+    MainViewModel mainViewModel = mainViewModelProvider.getMainViewModel(e.getProject());
+    CourseViewModel courseViewModel = mainViewModel.courseViewModel.get();
+    Authentication authentication = mainViewModel.authentication.get();
+    ExercisesTreeViewModel exercisesViewModel = mainViewModel.exercisesViewModel.get();
+    boolean isTutorialSelected =
+            exercisesViewModel != null
+            && authentication != null && courseViewModel != null
+            && exercisesViewModel.getSelectedItem() != null
+            && !(exercisesViewModel.getSelectedItem() instanceof ExerciseGroupViewModel)
+            && exercisesViewModel.findSelected().getLevel(2) != null
+            && ExerciseViewModel.Status.TUTORIAL.equals(
+               ((ExerciseViewModel) exercisesViewModel.findSelected().getLevel(2)).getStatus());
```

Comment:
Nit: I may have said this before, but this and the similar code in `confirmStart` could be put together to avoid duplication.